CONTRIBUTING.md

```diff
@@ -20,6 +21,46 @@ More resources can be found in the nascent [developer documentation](devdocs/REA
 - [Platform support policy](devdocs/PLATFORM_SUPPORT.md)
 - [Our Rust style](devdocs/rust_style.md)
 
+## Tips for submitting PRs
+
+Please talk to the core team before making major changes! See the [proposing design changes](#proposing-design-changes) for more details.
+
+### Linking and mentioning issues
+
+If your PR closes one or more issues, you can automatically link the PR with them by using one of the [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword):
+
+- This PR should close #xxxx
+- Fixes #xxxx
+
+You can also mention related issues, PRs or discussions!
+
+### Release notes section
+
+In our PR template, we have a "Release notes summary" section which will be included in our release notes for our blog.
+
+This section should include any information about your change which is relevant to a user of Nushell. You should try to keep it brief and simple to understand, and focus on the ways your change directly impacts the user experience. We highly encourage adding examples and, when relevant, screenshots in this section.
+
+This section might not be relevant for all PRs. If your PR is a work in progress, feel free to write "WIP"/"TODO"/etc in this section. You can also write "N/A" if this is a technical change which doesn't impact the user experience.
+
+If you're not sure what to put here, or need some help, a core team member would be glad to help you out. We may also makes some tweaks to your release notes section. Please don't take it personally, we just want to make sure our release notes are polished and easy to understand. Once the release notes section is ready, we'll add the (TODO label name) label to indicate that the release notes section is ready to be included in the actual release notes.
+
+### Tests and formatting checks
+
+Our CI system automatically checks formatting and runs our tests. If you're running into an issue, or just want to make sure everything is ready to go before creating your PR, you can run the checks yourself:
+
+- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
+- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used` to check that you're using the standard code style
+- `cargo test --workspace` to check that all tests pass (on Windows make sure to enable [developer mode](https://learn.microsoft.com/en-us/windows/apps/get-started/developer-mode-features-and-debugging))
+- `cargo run -- -c "use toolkit.nu; toolkit test stdlib"` to run the tests for the standard library
+
+Furthermore, you can also runs these checks all together using the `toolkit` module within Nushell as follows:
+```nushell
+use toolkit.nu # or use an `env_change` hook to activate it automatically
+toolkit check
+```
```

Comment:
I would put the toolkit as the more general newbie solution ahead of the specific commands. If you want to be sure and don't know your way around just use that. And then the details if you want to learn more or are already familiar with more of cargo. (Also we have some commands in later sections of t he same document. might be worth deduping)